### PR TITLE
Rearrange logic for consistency between cluster and meta markers

### PR DESCRIPTION
**Pull request by pinin4fjords**
This is a PR associated with https://github.com/ebi-gene-expression-group/scxa-workflows/pull/21, to bundle marker genes for metadata variable in a manner more consistent with that of clustering, since those markers are now generated by the same collection process in Galaxy, and named in the same way. 

Tested and works on experiments with and without metadata (i.e. cell type) markers.

### CODE REVIEW

File: bin/makeMarkerStats.R [hunk 1]
```diff
@@ -136,26 +120,61 @@ colnames(counts) <- colData(counts)$Barcode
...
-
-# Add in cell markers where provided
+all_marker_files <- unlist(lapply(c('cluster', 'meta'), function(markerType){
+    
+    if (! is.na(opt[[paste0(markerType, '_markers_dir')]])){
+        marker_files <- list.files(path = paste0(markerType, '_markers'), pattern = opt[[paste0(markerType, '_markers_file_pattern')]], full.names = TRUE)
+        
+        if (length(marker_files) > 0){
+
+            marker_cell_groupings <- gsub('_', ' ', sub('markers_(.+).tsv', '\\1', basename(marker_files)))
```
Comment (a-solovyev12) [inline bin/makeMarkerStats.R:130]: not sure what this line is doing?

Comment (pinin4fjords) [reply bin/makeMarkerStats.R:130]: If you have a file name like /path/to/markers_thing.tsv, it's extracting 'thing'.

File: main.nf [hunk 2]
```diff
@@ -638,46 +637,35 @@ FINAL_CLUSTERS.into{
...
 
     executor 'local'
     
     input:
-        file markersFile from SCANPY_CLUSTER_MARKERS
+        file markersFile from SCANPY_MARKERS
 
     output:
-        set stdout, file (markersFile) into CLUSTER_MARKERS_BY_RESOLUTION 
+        set stdout, file ('cluster_markers.tsv') optional true into CLUSTER_MARKERS_BY_RESOLUTION
```
Comment (a-solovyev12) [inline main.nf:648]: what happens downstream if no output is produced?

File: main.nf [hunk 3]
```diff
@@ -733,19 +790,14 @@ process bundle_summary {
...
-
     for matrix_type in filtered_normalised tpm_filtered; do
         if [ -d \${matrix_type}_dir ]; then
             makeMarkerStats.R \
                 --counts-dir=\${matrix_type}_dir \
                 --clusters-file=${clusters} \
-                --cluster-markers-dir=\$(pwd) \
-                --celltype-fields=${params.cellTypeField} \
-                --cellgroups-file=${cellMeta} \$celltype_markers_opt \
+                --cluster-markers-dir=cluster_markers \
```
Comment (a-solovyev12) [inline main.nf:798]: just out of curiosity, why are we using `=` here?

File: main.nf (hunk 2 above)
Comment (pinin4fjords) [reply main.nf:648]: The summary step would likely fail, triggering an error at the top level which would need to be investigated.

File: main.nf (hunk 3 above)
Comment (pinin4fjords) [reply main.nf:798]: Hmm, maybe that isn't necessary. But it works, so clearly allowed syntax.

**Review by a-solovyev12 — APPROVED** [review]
Looks fine to me!

**Comment by pinin4fjords** [comment]
Thanks @a-solovyev12 !

**Pull request merged**